TypeSystem/Schema/schema.h

```diff
@@ -1115,6 +1117,202 @@ struct LanguageSyntaxImpl<Language::JSON> final {
   // LCOV_EXCL_STOP
 };
 
+template <>
+struct LanguageSyntaxImpl<Language::TypeScript> final {
+  static std::string Header(const std::string&) {
+    // TODO(sompylasar): Implement common C++ and Current types in TypeScript.
+    return (
```

Comment:
Yes, and better be a published one. In this shape these declarations may cause conflicts, this needs to be tested, too.

I've just realized that the schema could also be installable, but maybe it's an overhead for now.